### PR TITLE
Improve tag-details & use some compositions

### PR DESCRIPTION
**Pull request by anbraten**
Wait for #359 to be merged first. <-- **Is merged**

Done
- refactored code a bit to use some generic compositions
- changed the way of tag-details loading the edit-image / edit-track screens by changing the url.

### CODE REVIEW

**Review by tcoro — COMMENTED** [review]
why didnt you add my tagdetails changes from https://github.com/project-leek/project-leek/issues/312

**Comment by anbraten** [comment]
> why didnt you add my tagdetails changes from #312

#312 so PR #322? That was about the image selection. IMO I haven't touched this here.

Or did you mean the UI improvements from #321? I currently moving your changes of that to a new separate PR.

**Comment by tcoro** [comment]
> > why didnt you add my tagdetails changes from #312
> 
> #312 so PR #322? That was about the image selection. IMO I haven't touched this here.
> 
> Or did you mean the UI improvements from #321? I currently moving your changes of that to a new separate PR.

ah okay, because the title says "improve tag details".
But then you get my

**Review by tcoro — APPROVED** [review]
... approval

**Comment by anbraten** [comment]
> ... approval

Ahh, yeah a bit misleading. I mainly introduced some the compositions and did some changes to the way the different screen in tag-edit-details are loaded.

File: packages/app/src/components/tag/TagStepImage.vue [hunk 1]
```diff
@@ -77,7 +81,7 @@ export default defineComponent({
     });
 
     onBeforeMount(async () => {
-      const track = await feathers.service('spotify-tracks').get(props.nfcTag.trackUri);
+      const track = await getTrackOfTag(props.nfcTag);
```
Comment (schnickalodeon) [inline packages/app/src/components/tag/TagStepImage.vue:84]: nice to see that my approch is used :D :D

File: packages/app/src/compositions/useNfcTag.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+export const getIsNfcTagValid = (nfcTag: Ref<NFCTag | undefined>): ComputedRef<boolean> =>
+  computed(() => {
+    if (!nfcTag.value) {
+      return false;
+    }
+
+    const tag = nfcTag.value;
+    return !!(tag.nfcData && tag.name && tag.group && tag.trackUri && tag.imageUrl);
+  });
```
Comment (schnickalodeon) [inline packages/app/src/compositions/useNfcTag.ts:63]: good idea 👍

**Review by schnickalodeon — APPROVED** [review]
lgtm 👍

File: packages/app/src/components/tag/TagStepImage.vue (hunk 1 above)
Comment (anbraten) [reply packages/app/src/components/tag/TagStepImage.vue:84]: 😉

**Pull request merged**